Kernel/KBOComparator.cpp

```diff
@@ -81,26 +89,34 @@ KBOComparator::KBOComparator(TermList tl1, TermList tl2, const KBO& kbo)
 
     // if both are proper terms, we calculate
     // weight and variable balances first
-    DHMap<unsigned,int> vars;
-    int w = 0;
-    countSymbols(kbo, vars, w, lhs, 1);
-    countSymbols(kbo, vars, w, rhs, -1);
 
     // we only care about the non-zero weights and counts
     bool varInbalance = false;
-    DHMap<unsigned,int>::Iterator vit(vars);
+    auto state = kbo._state;
```

Comment:
Given the following shenanigans maybe we should add a static `std::is_trivially_copyable` assertion to `KBO::State`.

Reply:
Note that `kbo._state` is a pointer, so hopefully nothing is copied?

Reply:
Ah, right - so maybe we should make it immovable or something. :thinking:  But not important!